controllers/admin/AdminReferrersController.php

```diff
@@ -179,49 +178,40 @@ public function renderForm()
-                    'name' => 'name',
-                    'required' => true,
-                    'autocomplete' => false,
+        $this->fields_form[0] = [
+            'form' => [
+                'legend' => [
+                    'title' => $this->trans('Affiliate', [], 'Admin.Shopparameters.Feature'),
+                    'icon' => 'icon-group',
                 ],
-                [
-                    'type' => 'password',
-                    'label' => $this->trans('Password', [], 'Admin.Global'),
-                    'name' => 'passwd',
-                    'desc' => $this->trans('Leave blank if no change.', [], 'Admin.Shopparameters.Help'),
-                    'autocomplete' => false,
-                ],
-            ],
-            'submit' => ['title' => $this->trans('Save', [], 'Admin.Actions')],
-        ]];
-
-        $moduleManagerBuilder = ModuleManagerBuilder::getInstance();
-        $moduleManager = $moduleManagerBuilder->build();
-
-        if ($moduleManager->isInstalled('trackingfront')) {
-            $this->fields_form[0]['form']['desc'] = [
-                $this->trans('Affiliates can access their data with this name and password.', [], 'Admin.Shopparameters.Feature'),
-                $this->trans('Front access:', [], 'Admin.Shopparameters.Feature') . ' <a class="btn btn-link" href="' . $uri . 'modules/trackingfront/stats.php" onclick="return !window.open(this.href);"><i class="icon-external-link-sign"></i> ' . $uri . 'modules/trackingfront/stats.php</a>',
-            ];
-        } else {
-            $this->fields_form[0]['form']['desc'] = [
-                $this->trans(
-                    'Please install the "%modulename%" module in order to give your affiliates access to their own statistics.',
+                'input' => [
                     [
-                        '%modulename%' => Module::getModuleName('trackingfront'),
+                        'type' => 'text',
+                        'label' => $this->trans('Name', [], 'Admin.Global'),
+                        'name' => 'name',
+                        'required' => true,
+                        'autocomplete' => false,
                     ],
-                    'Admin.Shopparameters.Notification'
-                ),
-            ];
-        }
+                    [
+                        'type' => 'password',
+                        'label' => $this->trans('Password', [], 'Admin.Global'),
+                        'name' => 'passwd',
+                        'desc' => $this->trans('Leave blank if no change.', [], 'Admin.Shopparameters.Help'),
+                        'autocomplete' => false,
+                    ],
+                ],
+                'submit' => ['title' => $this->trans('Save', [], 'Admin.Actions')],
+                'desc' => [
+                    $this->trans(
+                        'Please install the "%modulename%" module in order to give your affiliates access to their own statistics.',
```

Comment:
It doesn't really remove the link with this module since it actually encourages to install it... The thing changed here is that it will **always** display the suggestion to install the module even if it's installed

Besides if the module is absent it tries to get its name but in the absence of the module (and specifically its config files) the `Module::getModuleName` function is not able to display the module name and will fallback to `Module trackingfront`

Reply:
If the purpose is to get rid of any link to this external module then the whole description should actually be removed But this raises the question: if the description is not appropriate anymore maybe the form inputs themselves are not either? What is this form actually supposed to do?